### PR TITLE
docs: release management

### PR DESCRIPTION
**Pull request by mthpvg**
### Planning
https://github.com/upfrontIO/livingdocs-planning/issues/1291

### Communication
@DaRaFF the first draf. Please comment at will :)

### CODE REVIEW

File: setup-and-deployment/release/index.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# When to perform a release
+
+## Sprint from week 1 to 4
+| Week | Monday              | Midweek        | Friday                    |
+| ---- |:-------------------:| :-------------:| :-----------------------: |
+| 1    |                     |                |                           |
+| 2    |                     |                |                           |
+| 3    |                     |                |                           |
+| 4    | upstream releases   | testing *      |                           |
+| 5    | downstream releases | testing **     | handing upstream releases |
```
Comment (mthpvg) [inline setup-and-deployment/release/index.md]: Add merging time.

File: setup-and-deployment/release/index.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- [livingdocs-server](https://github.com/upfrontIO/livingdocs-server)
+- [livingdocs-editor](https://github.com/upfrontIO/livingdocs-editor)
+
+### Assumptions
+
+- The last release and tag of the upstream application is `v2.1.0`.
+
+### Steps
+
+1. `git checkout -b release-bump-2.2.0`
```
Comment (DaRaFF) [inline setup-and-deployment/release/index.md]: I would declare step 1-5 as optional. It's only needed when no minor or major bump version already exists.

File: setup-and-deployment/release/index.md (hunk 1 above)
Comment (mthpvg) [reply setup-and-deployment/release/index.md]: explicit days and time

File: setup-and-deployment/release/index.md (hunk 2 above)
Comment (DaRaFF) [reply setup-and-deployment/release/index.md]: I also recommend to add a script to support the steps 1-5.

File: setup-and-deployment/release/index.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- The latest maintenance tag of the corresponding upstream application is `2.1.0`.
+
+### Steps
+
+1. `git checkout -b update-upstream-2.1.0`
+2. Update the upstream project to its latest maintenance release, from our previous example it is `2.1.0` in the downstream *package.json*.
+3. `git commit -m "feat: update upstream to version 2.1.0"`
+4. `git push origin update-upstream-2.1.0`
+5. `git branch -D update-upstream-2.1.0`
+6. Merge the Pull request in the Github UI and wait for the semantic release script to produce the new `v17.5.0` release
```
Comment (DaRaFF) [inline setup-and-deployment/release/index.md]: For the moment we have disabled semantic-release in our downstream. So we don't need step 1-6.

File: setup-and-deployment/release/index.md (hunk 1 above)
Comment (DaRaFF) [reply setup-and-deployment/release/index.md]: I would try to explicitly describe *when* (day and time) and *why* you are doing a specific steps. You can also add a section where you describe when you can abandon this rule.

**Comment by mthpvg** [comment]
@DaRaFF review round 2

File: reference-docs/maintenance/release-management/workflow.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+### Assumptions
+
+- The latest, manual, tag of the downstream application is `4.0.1`.
+
+### Steps
+
+1. `./node_modules/@livingdocs/release-tools/li-release init-release 5.0.0 master`
+2. `./node_modules/@livingdocs/release-tools/li-release finish-release 5.0.0`
+
+# When to perform a release
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/workflow.md]: I would start in the document with when to perform a release and make a link to how to make a release (you already have marked the how with **bold** letters in this section.

File: reference-docs/maintenance/release-management/workflow.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+A sprint has 4 weeks, the release workflow happens in the last two weeks:
+
+- Week 3
+  - Monday 10am
+    - **Open the usptream releases**: it marks the beginning of the integration week (testing is done by the developers), only fix commits are to be added to the release branches.
+  - Friday 10am
+    - **Close the upstream releases**: it marks the end of the integration week, no commits are to be added to the release branches anymore.
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/workflow.md]: What I miss here is:
- How and when are the release notes put together
- Do we already know how to provide the info of a new release to our customers?

File: reference-docs/maintenance/release-management/workflow.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+There are multiple downstreams, for the NZZ, Bluewin and our own Service:
+- [livingdocs-service-server](https://github.com/upfrontIO/livingdocs-service-server)
+- [livingdocs-service-editor](https://github.com/upfrontIO/livingdocs-service-editor)
+
+### Assumptions
+
+- The latest, manual, tag of the downstream application is `4.0.1`.
+
+### Steps
+
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/workflow.md]: you should write that for the moment we are independent from semantic-release in downstreams AND we always do a major bump after every sprint.

File: reference-docs/maintenance/release-management/workflow.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+4. `git branch -D release-bump-2.2.0`
+5. Merge the Pull request in the Github UI and wait for the semantic release script to produce the new `v2.2.0` release and tag.
+
+**Actual release step**:
+
+1. `./node_modules/@livingdocs/release-tools/li-release create-maintenance-release --base-tag 2.1.0 --npm-auth-token <token>`
+
+## Downstream applications
+
+There are multiple downstreams, for the NZZ, Bluewin and our own Service:
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/workflow.md]: You should mention that they handle their downstream release on their own.
How do we handle our boilerplate? We should respect the boilerplate too, as same as we do it for the service.

File: reference-docs/maintenance/release-management/workflow.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+- [livingdocs-server](https://github.com/upfrontIO/livingdocs-server)
+- [livingdocs-editor](https://github.com/upfrontIO/livingdocs-editor)
+
+### Assumptions
+
+- The release use `v2.1.0` as a base tag.
+
+### Steps
+
+**If** `v2.1.0` is not the latest tag/release and to avoid potential versioning conflicts one need to execute:
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/workflow.md]: I would try to express better that this step is optional.

**Comment by mthpvg** [comment]
@DaRaFF Round 3, bring em in!

File: reference-docs/maintenance/release-management/when.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# When to perform a release
+
+A sprint has 4 weeks, the release workflow happens in the last two weeks:
+
+- Week 3
+  - Monday 10am
+    - Open the usptream releases: it marks the beginning of the integration week (testing is done by the developers), only fix commits are to be added to the release branches.
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/when.md]: upstream

Comment (DaRaFF) [reply reference-docs/maintenance/release-management/when.md]: only fix commits are allowed on release branches.

File: reference-docs/maintenance/release-management/when.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    - Create the release notes in [the release notes repository](https://github.com/upfrontIO/livingdocs-release-notes)
+  - Friday 10am
+    - Close the upstream releases: it marks the end of the integration week, no commits are to be added to the release branches anymore.
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/when.md]: no more commits are allowed on the release branches.

File: reference-docs/maintenance/release-management/how.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+Clients who develop their downstreams on their own handle releases by themselves:
+- NZZ
+- Bluewin
+
+Livingdocs handles the releases of its two downstreams :
+- [livingdocs-service-server](https://github.com/upfrontIO/livingdocs-service-server)
+- [livingdocs-service-editor](https://github.com/upfrontIO/livingdocs-service-editor)
+- [livingdocs-server-boilerplate](https://github.com/upfrontIO/livingdocs-server-boilerplate)
+- [livingdocs-editor-boilerplate]( (https://github.com/upfrontIO/livingdocs-editor-boilerplate)
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/how.md]: the link is wrong - you have 2 opening braces in the link

File: reference-docs/maintenance/release-management/how.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+#### Mandatory
+
+1. `./node_modules/@livingdocs/release-tools/li-release create-maintenance-release --base-tag 2.1.0 --npm-auth-token <token>`
+
+## Downstream applications
+
+Clients who develop their downstreams on their own handle releases by themselves:
+- NZZ
+- Bluewin
+
```
Comment (DaRaFF) [inline reference-docs/maintenance/release-management/how.md]: aren't there more like SZ, ippen, ...?

File: reference-docs/maintenance/release-management/when.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
+# When to perform a release
+
+A sprint has 4 weeks, the release workflow happens in the last two weeks:
+
+- Week 3
+  - Monday 10am
+    - Open the upstream releases: it marks the beginning of the integration week (testing is done by the developers), only fix commits are allowed on release branches.
```
Comment (rebelact) [inline reference-docs/maintenance/release-management/when.md]: @mthpvg isn't the integration week the last week of the sprint (the 4th week)?

Comment (mthpvg) [reply reference-docs/maintenance/release-management/when.md]: Off by one error :).

**Pull request merged**